doc/OL_preflight_check.py

```diff
@@ -0,0 +1,358 @@
+from airflow import __version__ as airflow_version
+from airflow.utils.trigger_rule import TriggerRule
+from distutils.version import StrictVersion
+
+LINEAGE_BACKEND = "MARQUEZ"  # update this if using any other backend for OpenLineage events ingestion and implement custom checks in other_server_check function
+BYPASS_LATEST_VERSION_CHECK = False  # set this to True to bypass the latest version check for openlineage-airflow library and OpenLineage provider package
+MARQUEZ_PATH_PREFIX = "/api/v1/"
+
+log = logging.getLogger(__name__)
+
+def parse_prefix(version_string):
+    """
+    Parse the version string and return the prefix.
+    This function is used to extract the base version (major.minor.patch) from a
+    version string possibly containing additional metadata.
+    :param version_string: str, version string (e.g., '2.7.0+astro1.2')
+    :return: str or None, base version (e.g., '2.7.0') if match found, otherwise None
+    """
+    log.debug(f"Airflow Version string: {version_string} ")
+    pattern = r'^(\d+\.\d+\.\d+)'
+    match = re.match(pattern, version_string)
+    return match.group(1) if match else None
+
+
+def check_ol_enabled():
+    """
+    This function is used to check if OpenLineage is disabled using environment variables.
+    """
+    if os.environ.get('AIRFLOW__OPENLINEAGE__DISABLED', 'false').lower() == 'true':
+        raise Exception("OpenLineage is disabled due to the environment variable AIRFLOW__OPENLINEAGE__DISABLED")
+    elif os.environ.get('OPENLINEAGE_DISABLED', 'false').lower() == 'true':
+        raise Exception("OpenLineage is disabled due to the environment variable OPENLINEAGE_DISABLED")
+    return
+
+
+def check_airflow_version():
+    """
+    This function is used to check the airflow version and return the branch based on the version.
+    <2.1: airflow_version_unsupported
+    >=2.1 and <2.7: uses openlineage-airflow library
+    >=2.7: uses provider package
+    returns the branch based on the version.
+    """
+    try:
+        version = StrictVersion(parse_prefix(airflow_version))
+    except ValueError:
+        raise ValueError("Failed to parse airflow version")
+    if version < StrictVersion('2.1'):
+        raise Exception('OpenLineage is not supported in Airflow versions <2.1')
+    elif version < StrictVersion('2.7'):
+        return 'OL_library'
+    else:
+        return 'provider_package'
+
+def OL_library_config(**kwargs):
+    """
+    This function is used to check the OL configuration when using openlineage-airflow library.
+    This function also ensures that the openlineage-airflow library is installed and up-to-date.
+    pushes the OL_type to xcom for other tasks to refer.
+    returns the branch based on the config type.
+    :param kwargs:
+    """
+    log.info("Airflow version is >=2.1 and <2.7, checking openlineage-airflow library configuration")
+    library_name = "openlineage-airflow"
+    check_library_version(library_name, get_latest_version_pypi(library_name))
```

Comment:
What if there was an error in getting latest version and we got `None` ?

Reply:
Handled it now. It will now give a warning and skip this check.